hbase-hbtop/src/main/java/org/apache/hadoop/hbase/hbtop/mode/RegionModeStrategy.java

```diff
@@ -0,0 +1,183 @@
+      startCode = Bytes.toString(elements[2]);
+      replicaId = elements.length == 4 ?
+        Integer.valueOf(Bytes.toString(elements[3])).toString() : "";
+      region = RegionInfo.encodeRegionName(regionMetrics.getRegionName());
+    } catch (IOException ignored) {
+    }
+
+    ret.put(Field.NAMESPACE, namespaceName);
+    ret.put(Field.TABLE, tableName);
+    ret.put(Field.START_CODE, startCode);
+    ret.put(Field.REPLICA_ID, replicaId);
+    ret.put(Field.REGION, region);
+    ret.put(Field.START_KEY, startKey);
+    ret.put(Field.REGION_SERVER, serverMetrics.getServerName().toShortString());
+    ret.put(Field.LONG_REGION_SERVER, serverMetrics.getServerName().getServerName());
+
+    RequestCountPerSecond requestCountPerSecond = requestCountPerSecondMap.get(regionName);
+    if (requestCountPerSecond == null) {
+      requestCountPerSecond = new RequestCountPerSecond();
+      requestCountPerSecondMap.put(regionName, requestCountPerSecond);
+    }
+    requestCountPerSecond.refresh(lastReportTimestamp, regionMetrics.getReadRequestCount(),
+      regionMetrics.getFilteredReadRequestCount(), regionMetrics.getWriteRequestCount());
+
+    ret.put(Field.READ_REQUEST_COUNT_PER_SECOND,
+      requestCountPerSecond.getReadRequestCountPerSecond());
+    ret.put(Field.FILTERED_READ_REQUEST_COUNT_PER_SECOND,
+        requestCountPerSecond.getFilteredReadRequestCountPerSecond());
+    ret.put(Field.WRITE_REQUEST_COUNT_PER_SECOND,
+      requestCountPerSecond.getWriteRequestCountPerSecond());
+    ret.put(Field.REQUEST_COUNT_PER_SECOND,
+      requestCountPerSecond.getRequestCountPerSecond());
+
+    ret.put(Field.STORE_FILE_SIZE, regionMetrics.getStoreFileSize());
+    ret.put(Field.UNCOMPRESSED_STORE_FILE_SIZE, regionMetrics.getUncompressedStoreFileSize());
+    ret.put(Field.NUM_STORE_FILES, regionMetrics.getStoreFileCount());
+    ret.put(Field.MEM_STORE_SIZE, regionMetrics.getMemStoreSize());
+    ret.put(Field.LOCALITY, regionMetrics.getDataLocality());
+
+    long compactingCellCount = regionMetrics.getCompactingCellCount();
+    long compactedCellCount = regionMetrics.getCompactedCellCount();
+    float compactionProgress = 0;
+    if  (compactedCellCount > 0) {
+      compactionProgress = 100 * ((float) compactedCellCount / compactingCellCount);
+    }
+
+    ret.put(Field.COMPACTING_CELL_COUNT, compactingCellCount);
+    ret.put(Field.COMPACTED_CELL_COUNT, compactedCellCount);
+    ret.put(Field.COMPACTION_PROGRESS, compactionProgress);
+
+    FastDateFormat df = FastDateFormat.getInstance("yyyy-MM-dd HH:mm:ss");
+    long lastMajorCompactionTimestamp = regionMetrics.getLastMajorCompactionTimestamp();
+
+    ret.put(Field.LAST_MAJOR_COMPACTION_TIME,
+      lastMajorCompactionTimestamp == 0 ? "" : df.format(lastMajorCompactionTimestamp));
+
+    return ret;
+  }
+
+  @Nullable
+  @Override
+  public DrillDownInfo drillDown(Record selectedRecord) {
+    // do nothing
+    return null;
+  }
+}
```

Comment:
You couldn't use RegionMetrics because you need to use your Record type constructing views?

Reply:
Yes. I don't use RegionMetrics directly because, as you said, some work is needed to construct views. I build the Record object from ServerMetrics and RegionMetrics.

Reply:
Is this then a commentary on how poorly Region and ServerMetrics are done? They don't aggregate or compose as your Record object does? We should move toward your model doing internal metrics?

Reply:
I don't think so. We have 4 modes in hbtop as of now, so I need an abstraction for metric records and it's this Record class. I don't think we need to modify ServerMetrics and RegionMetrics.